packages/gatsby/src/redux/machines/page-component.ts

```diff
@@ -96,8 +133,8 @@ module.exports = Machine(
         }, 0)
       },
       setQuery: assign({
-        query: (ctx, event) => {
-          if (typeof event.query !== `undefined` || event.query !== null) {
+        query: (ctx, event): string => {
+          if (typeof event.query !== `undefined` && event.query !== null) {
```

Comment:
This looks correct now actually - either I miss something and previous condition always was satisfied (and we always use `return event.query` path) or there is weird nuance in here

Reply:
Yup was confused but now it's safer